author: Andres Salomon <dilinger@debian.org>

libdav1d added some asm stuff that broke at least with clang-16:

https://buildd.debian.org/status/fetch.php?pkg=chromium&arch=arm64&ver=124.0.6367.60-1&stamp=1713551216&raw=0

../../third_party/dav1d/libdav1d/src/arm/asm.S:75:17: error: unknown architectural extension: dotprod
../../third_party/dav1d/libdav1d/src/arm/asm.S:76:17: error: unknown architectural extension: i8mm

I'm not clear is this is a clang-16 issue (and dotprod/i8mm is supported
in arm64 in later clang assemblers), or if upstream is targeting higher
baseline architecture and we'll always need to remove this. Needs more
research (ie, bugging the #debian-arm folks).

--- a/third_party/dav1d/config/linux/arm64/config.h
+++ b/third_party/dav1d/config/linux/arm64/config.h
@@ -41,9 +41,9 @@
 
 #define HAVE_ASM 1
 
-#define HAVE_AS_ARCHEXT_DOTPROD_DIRECTIVE 1
+#define HAVE_AS_ARCHEXT_DOTPROD_DIRECTIVE 0
 
-#define HAVE_AS_ARCHEXT_I8MM_DIRECTIVE 1
+#define HAVE_AS_ARCHEXT_I8MM_DIRECTIVE 0
 
 #define HAVE_AS_ARCHEXT_SVE2_DIRECTIVE 1
 
@@ -59,11 +59,11 @@
 
 #define HAVE_DLSYM 1
 
-#define HAVE_DOTPROD 1
+#define HAVE_DOTPROD 0
 
 #define HAVE_GETAUXVAL 1
 
-#define HAVE_I8MM 1
+#define HAVE_I8MM 0
 
 #define HAVE_POSIX_MEMALIGN 1
 
